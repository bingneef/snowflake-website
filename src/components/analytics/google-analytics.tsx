function GoogleAnalytics() {
  return (
    <>
      <script
        async={true}
        src="https://www.googletagmanager.com/gtag/js?id=UA-77301580-1"
      />
      <script
        dangerouslySetInnerHTML={{
          __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());

            gtag('config', 'UA-77301580-1');
          `
        }}
      />
    </>
  );
}

export default GoogleAnalytics;
