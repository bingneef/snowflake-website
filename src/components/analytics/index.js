/* PLOP_INJECT_IMPORT */
import FullStory from "./full-story";
import GoogleAnalytics from "./google-analytics";
import Hotjar from "./hotjar";

export {
  /* PLOP_INJECT_EXPORT */
  FullStory,
  GoogleAnalytics,
  Hotjar
};
