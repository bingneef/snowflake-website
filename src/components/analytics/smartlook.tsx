export function sendEvent(eventName: string, eventProperties?: {}) {
  if (typeof smartlook === "undefined") {
    // tslint:disable-next-line:no-console
    console.log(
      `Not sending event: ${eventName}, not in production environment`
    );
  } else {
    smartlook("track", eventName, eventProperties);
  }
}

function Smartlook() {
  if (process.env.NODE_ENV !== "production") {
    return null;
  }

  return (
    <script
      dangerouslySetInnerHTML={{
        __html: `
          window.smartlook||(function(d) {
          var o=smartlook=function(){ o.api.push(arguments)},h=d.getElementsByTagName('head')[0];
          var c=d.createElement('script');o.api=new Array();c.async=true;c.type='text/javascript';
          c.charset='utf-8';c.src='https://rec.smartlook.com/recorder.js';h.appendChild(c);
          })(document);
          smartlook('init', '1918e07593de805de944505d6e91fb8ceec2b696');
          `
      }}
    />
  );
}

export default Smartlook;
