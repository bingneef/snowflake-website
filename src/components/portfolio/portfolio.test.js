import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { render, fireEvent, cleanup } from "react-testing-library";
import * as fullStory from "../analytics/full-story";
import * as mixpanel from "../analytics/mixpanel";

// add custom jest matchers from jest-dom
import "jest-dom/extend-expect";

import Portfolio from "./portfolio";

describe("Portfolio", () => {
  beforeEach(cleanup);

  it("renders without error", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Portfolio />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("matches the snapshot", () => {
    const { container } = render(<Portfolio />);
    expect(container).toMatchSnapshot();
  });

  it("sends analytics event on item click", () => {
    const { getByTestId } = render(<Portfolio />);
    mixpanel.sendEvent = jest.fn();
    fullStory.sendEvent = jest.fn();

    const item = getByTestId("portfolio-item-1");
    fireEvent.click(item);

    const payload = [
      "Portfolio OpenItem",
      {
        tag: item.getAttribute("data-item-tag")
      }
    ];
    expect(mixpanel.sendEvent).toHaveBeenCalledWith(...payload);
    expect(fullStory.sendEvent).toHaveBeenCalledWith(...payload);
  });

  it("sends analytics event on tab click", () => {
    const { getByTestId } = render(<Portfolio />);
    const tabs = ["serious", "tools", "fun"];

    for (let tab of tabs) {
      mixpanel.sendEvent = jest.fn();
      fullStory.sendEvent = jest.fn();

      const item = getByTestId(`tab-${tab}`);
      fireEvent.click(item);

      const payload = [
        "Portfolio OpenTab",
        {
          tab
        }
      ];
      expect(mixpanel.sendEvent).toHaveBeenCalledWith(...payload);
      expect(fullStory.sendEvent).toHaveBeenCalledWith(...payload);

      expect(item.getAttribute("class").includes("is-active"));
    }
  });

  it("does not crash when the tab has is not in the json", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Portfolio initialTab={"bogus"} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
