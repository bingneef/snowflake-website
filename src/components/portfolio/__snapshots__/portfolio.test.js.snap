// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio matches the snapshot 1`] = `
<section
  className="section is-info root"
  id="portfolio"
>
  <div
    className="container"
  >
    <div
      className="heading"
    >
      <h1
        className="title"
      >
        Portfolio
      </h1>
      <h2
        className="subtitle"
      >
        Its all about that 
        <strong>
          fluffiness
        </strong>
        .
      </h2>
    </div>
    <div
      className="tabs"
    >
      <ul>
        <li
          className="is-active"
          onClick={[Function]}
        >
          <a>
            <span
              className="icon"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-rocket fa-w-16 "
                data-icon="rocket"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505.05 19.1a15.89 15.89 0 0 0-12.2-12.2C460.65 0 435.46 0 410.36 0c-103.2 0-165.1 55.2-211.29 128H94.87A48 48 0 0 0 52 154.49l-49.42 98.8A24 24 0 0 0 24.07 288h103.77l-22.47 22.47a32 32 0 0 0 0 45.25l50.9 50.91a32 32 0 0 0 45.26 0L224 384.16V488a24 24 0 0 0 34.7 21.49l98.7-49.39a47.91 47.91 0 0 0 26.5-42.9V312.79c72.59-46.3 128-108.4 128-211.09.1-25.2.1-50.4-6.85-82.6zM384 168a40 40 0 1 1 40-40 40 40 0 0 1-40 40z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
            <span>
              Serious
            </span>
          </a>
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          <a>
            <span
              className="icon"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-wrench fa-w-16 "
                data-icon="wrench"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M507.73 109.1c-2.24-9.03-13.54-12.09-20.12-5.51l-74.36 74.36-67.88-11.31-11.31-67.88 74.36-74.36c6.62-6.62 3.43-17.9-5.66-20.16-47.38-11.74-99.55.91-136.58 37.93-39.64 39.64-50.55 97.1-34.05 147.2L18.74 402.76c-24.99 24.99-24.99 65.51 0 90.5 24.99 24.99 65.51 24.99 90.5 0l213.21-213.21c50.12 16.71 107.47 5.68 147.37-34.22 37.07-37.07 49.7-89.32 37.91-136.73zM64 472c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
            <span>
              Tools
            </span>
          </a>
        </li>
        <li
          className=""
          onClick={[Function]}
        >
          <a>
            <span
              className="icon"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-heart fa-w-16 "
                data-icon="heart"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </span>
            <span>
              Fun
            </span>
          </a>
        </li>
      </ul>
    </div>
    <div
      className="columns is-multiline"
    >
      <div
        className="column is-3"
        data-item-tag="snowflakewebsite"
        onClick={[Function]}
      >
        <div
          className="portfolio-item-container"
        >
          <img
            src="/static/assets/images/portfolio/snowflakewebsite-min.png"
          />
          <h2
            className="subtitle"
          >
            Snowflake Website
          </h2>
        </div>
      </div>
      <div
        className="column is-3"
        data-item-tag="theweddingapp"
        onClick={[Function]}
      >
        <div
          className="portfolio-item-container"
        >
          <img
            src="/static/assets/images/portfolio/theweddingapp-min.png"
          />
          <h2
            className="subtitle"
          >
            TheWeddingApp
          </h2>
        </div>
      </div>
      <div
        className="column is-3"
        data-item-tag="depunaise"
        onClick={[Function]}
      >
        <div
          className="portfolio-item-container"
        >
          <img
            src="/static/assets/images/portfolio/depunaise-min.png"
          />
          <h2
            className="subtitle"
          >
            De Punaise
          </h2>
        </div>
      </div>
      <div
        className="column is-3"
        data-item-tag="musicgenius"
        onClick={[Function]}
      >
        <div
          className="portfolio-item-container"
        >
          <img
            src="/static/assets/images/portfolio/empty.svg"
          />
          <h2
            className="subtitle"
          >
            MusicGenius
          </h2>
        </div>
      </div>
      <div
        className="column is-3"
        data-item-tag="sportscoach"
        onClick={[Function]}
      >
        <div
          className="portfolio-item-container"
        >
          <img
            src="/static/assets/images/portfolio/sportscoach-min.png"
          />
          <h2
            className="subtitle"
          >
            Sportscoach
          </h2>
        </div>
      </div>
      <div
        className="column is-3"
        data-item-tag="spot"
        onClick={[Function]}
      >
        <div
          className="portfolio-item-container"
        >
          <img
            src="/static/assets/images/portfolio/spot-min.png"
          />
          <h2
            className="subtitle"
          >
            Spot
          </h2>
        </div>
      </div>
    </div>
  </div>
</section>
`;
